sdk/nodejs/sns/topicPolicy.ts

```diff
@@ -163,5 +165,5 @@ export interface TopicPolicyArgs {
     /**
      * The fully-formed AWS policy as JSON.
      */
-    policy: pulumi.Input<string>;
+    policy: pulumi.Input<string | PolicyDocument>;
```

Comment:
This looks meaningfully better for the users.

Reply:
This is clearer, but still breaking. `args.policy = new PolicyDocument(...)` is IMO strictly better. `check(args.policy)` now requires `check` to handle multiple types, which is strictly worse.

On the whole, I think it's an improvement.

Reply:
Why is it breaking?

Reply:
> check(args.policy) now requires check to handle multiple types, which is strictly worse.

I think this is a listed reason, but which user programs are calling check()? Something using transforms or policy?

Reply:
`check` is a stand-in for any read. It's the same argument as https://github.com/pulumi/pulumi-aws/pull/3743#discussion_r1541540270, maybe worded a little less clearly.

Reply:
Gotcha. So more a theoretical possibility.